### ISSUE TITLE
EdgeRouting Drawn from Node Center Instead of Bounds Edge

### ISSUE THREAD

**Issue by emeeks** [tgdwyer/WebCola#119]
In trying to follow the Unix example of edge routing, I find my nodes are being drawn from and to what seem to be the node center instead of the edge of the bounds. I'm using cola.vpsc.makeEdgeBetween which seems to work fine during the dynamic phase of the graph for drawing edges from and to defined boundaries, and I don't see any specifics within the Unix example that indicate how to format or pass the proper bounds to the edge routing function.

![weirdrouteedges](https://cloud.githubusercontent.com/assets/495634/8269813/899303da-1770-11e5-8525-2a068413eb93.png)

**Comment by emeeks** [comment]
Oh, and it's using cardinal interpolation for the lines, but you can ignore that, the point is the start and terminus of each line is in an odd place.

**Comment by emeeks** [comment]
Here's an example that swaps out the unix example with a new dataset that reflects the above strange drawing behavior:

http://bl.ocks.org/emeeks/34cdccd1aebd04d09b35

**Comment by tgdwyer** [comment]
I found the problem.  The edge routing is expecting nodes to have an ```id``` property that is an integer value matching their position in the nodes array.  Is this good API design?  No!  It's horrible and I'm an evil person for making it that way.  I'll try to figure out a better way to do it, but in the meantime, if you add the following line somewhere before the routing:

            nodes.forEach(function (v, i) { v.id = i });

then you get a nicer picture:

![image](https://cloud.githubusercontent.com/assets/2754673/8304866/0d395d3c-19ed-11e5-9638-5c426f90b994.png)

note, I also increased the padding and the ideal edge length a bit:

                .flowLayout('x', 200)
                .jaccardLinkLengths(200,2);

there's still some bugs when the obstacles are close together, but the above works OK.
